enhancements/sdk-generate-commands.md

```diff
@@ -0,0 +1,353 @@
+
+## Goals
+
+* Simplify UX for generating bundle and package manifests formats.
+* Delineate bundle and package manifests format generation and usage.
+* _(New project layouts)_ Use a Makefile wherever possible.
+
+## Non-Goals
+
+* Deprecate or remove package manifests support from the Operator SDK.
+* Discuss the `run` subcommand in relation to bundles or package manifests.
+
+## Proposal
+
+There are two parts to this proposal:
+
+1. Add `generate bundle` and `generate packagemanifests` subcommands for both old and new project layouts.
+1. _(New project layouts)_ Add Makefile recipes `make bundle` and `make bundle-build`
+that invoke `generate bundle` and `docker build -f bundle.Dockerfile .`, respectively.
+
+### User Stories
+
+#### Story 1
+
+As an Operator developer of a new layout-style project, I want to generate a
+bundle format on disk using `kustomize` or from manifests on disk in a custom location.
+
+#### Story 2
+
+As an Operator developer of an old layout-style project, I want to generate a
+bundle format on disk using manifests existing in my `deploy` directory.
+
+#### Story 3
+
+As an Operator developer of a new layout-style project, I want to generate a
+package manifests format on disk using `kustomize` or from manifests on disk in
+a custom location.
+
+#### Story 4
+
+As an Operator developer of an old layout-style project, I want to generate a
+package manifests format on disk using manifests existing in my `deploy` directory.
+
+#### Story 5
+
+As a developer of a vanilla kubebuilder project, I would like documentation on how to add the bundle/package
+generation steps described here to my project.
+
+### Implementation Details/Notes/Constraints
+
+#### `kustomize` usage (new project layouts)
+
+Bundle and package manifests generation should leverage `kustomize` because manifests
+in `config` are meant to be configurable, i.e. for a particular deployment strategy.
+For example, `make deploy` will call `kustomize build` on the default `kustomization.yaml`
+to generate a working set of Operator manifests that is directly applied to a cluster.
+A bundle or package should contain manifests configured to be cluster-ready, therefore
+both `bundle` and `packagemanifests` subcommands must be able to consume manifests
+from stdin.
+
+New project layouts rely on `kustomize build` to generate cluster-ready manifests,
+the output of which written either to `kubectl apply -f -` or disk if desired.
+However, each command in each scenario should be able to handle direct input from
+manifests on disk. This mode is important for new project layouts in the case where the
+environment cannot use `kustomize` when generating either format, ex. in a pipeline.
+For old project layouts, this is the default. New project layouts,
+the default is to use `kustomize build config/default | operator-sdk generate <bundle|packagemanifests>`.
```

Comment:
Are we speaking bout to change the default makefile target or create a new one with? if it is a new one I am OK with.

Reply:
A new one.